player/command.c

```diff
@@ -5080,6 +5080,16 @@ int run_command(struct MPContext *mpctx, struct mp_cmd *cmd, struct mpv_node *re
         break;
     }
 
+    case MP_CMD_EXPAND_PATH: {
+        if (!res)
+            return -1;
+        *res = (mpv_node){
+            .format = MPV_FORMAT_STRING,
+            .u.string = mp_get_user_path(mpctx, mpctx->global, cmd->args[0].v.s)
```

Comment:
That's completely wrong btw.

Reply:
Blame @avih, I'm just copypaste-coding from `expand-text` here. What would the proper way be?

Reply:
It must be a talloc allocation with no parent. I don't know whether passing NULL as parent would do the right thing. The expand-text code is definitely broken.

Reply:
Nevermind, I'm dumb, expand-text is right because the function it calls takes different parameters; this one is squarely my fault.